File: Command/Python/src/executors/garage_door.py
class GarageDoor:

    def __init__(self, name: str) -> None:
        self._name = f"{name} door"
        self.speed = self.Speeds.HIGH

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self) -> None:
        raise AttributeError("'name' is read only")

    def up(self) -> None:
        print(f"{self._name} is open")

    def down(self) -> None:
        print(f"{self._name} is closed")

    def stop(self) -> None:
        print(f"{self._name} stopping action")

    def light_on(self) -> None:
        print(f"{self._name} light is on")

    def light_off(self) -> None:
        print(f"{self._name} light is off")

File: Command/DotNet/Command/executors/GarageDoor.cs
namespace CommandPattern.Executors;


public class GarageDoor
{
    public string Name { get; }

    public GarageDoor(string name)
    {
        this.Name = $"{name} door";
    }
    public void Up()
    {
        Console.WriteLine($"{this.Name} is open");
    }

    public void Down()
    {
        Console.WriteLine($"{this.Name} is closed");
    }

    public void Stop()
    {
        Console.WriteLine($"{this.Name} stopping action");
    }

    public void LightOn()
    {
        Console.WriteLine($"{this.Name} light is on");
    }

    public void LightOff()
    {
        Console.WriteLine($"{this.Name} light is off");
    }
}